From d0a333a3b9ed9184dba5c63b57c46e84af46941d Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankitja@vmware.com>
Date: Mon, 17 Oct 2022 10:58:25 +0000
Subject: [PATCH] installer: Adding support for /dev/disk/by-path

Usecase:
- 'disk' : '/dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:1:0'
Also, similar way 'disk' can be defined inside 'partition'
table as well.

Change-Id: I9f120582f34f295e59a831e7295dfcc2ff813f5c
Signed-off-by: Ankit Jain <ankitja@vmware.com>
---
 photon_installer/installer.py  | 7 +++++++
 photon_installer/ks_config.txt | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/photon_installer/installer.py b/photon_installer/installer.py
index b19f186..1d01139 100755
--- a/photon_installer/installer.py
+++ b/photon_installer/installer.py
@@ -1114,9 +1114,16 @@ class Installer(object):
         # 3) to create physical partition representation for VG
         vg_partitions = {}
 
+        # /dev/disk/by-path/pci-* -> ../../dev/sd* is symlink to device file
+        # To handle the case for ex:
+        # 'disk' : '/dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0'
+        self.install_config['disk'] = os.path.realpath(self.install_config['disk'])
+
         default_disk = self.install_config['disk']
         partitions = self.install_config['partitions']
         for partition in partitions:
+            if 'disk' in partition:
+                partition['disk'] = os.path.realpath(partition['disk'])
             disk = partition.get('disk', default_disk)
             if disk not in ptv:
                 ptv[disk] = []
diff --git a/photon_installer/ks_config.txt b/photon_installer/ks_config.txt
index dc3a1fa..335a019 100644
--- a/photon_installer/ks_config.txt
+++ b/photon_installer/ks_config.txt
@@ -57,6 +57,8 @@ Kickstart config file is a json format with following possible parameters:
 	Target"s disk device file path to install into, such as "/dev/sda".
 	Loop device is also supported.
 	Example: { "disk": "/dev/sdb" }
+	/dev/disk/by-path is also supported.
+	Example: { "disk": "/dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:1:0" }
 
 "eject_cdrom" (optional)
 	Eject or not cdrom after installation completed.
-- 
2.30.5

